### PR TITLE
Gravatar: Magic login improvements

### PR DESCRIPTION
**Pull request by wellyshen**
Related to 105749-Automattic/gravatar

## Proposed Changes

* Align the magic login page with the new design (IXT40Xck14laWK456R4wB2-fi)
* Bring the link of the password/social login page back
* Customize the links and the submit button of the WPJM login page
* Show the "Resend the verification email" function when `resendEmailCountdown` is `0` 
* Add track events:
  * `calypso_${ oauth2Client.name }_magic_login`
  * `calypso_${ oauth2Client.name }_magic_login_click_login_page_link`
  * `calypso_${ oauth2Client.name }_magic_login_click_resend_email`
  * `calypso_${ oauth2Client.name }_magic_login_click_use_different_email`

## Testing Instructions

* Check out this PR and run it
* Logout your account
* Go to the magic login page of [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D943e6fc69e6b8c3d5d488f7d9e3ee0083c5700a3136cc3f25205360ef89c825f%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1), and you will the design have been updated:

> Note: The `a` link of the "a WordPress.com account" will be implemented once it's ready.

| Gravatar | WPJM |
| - | - |
| 截圖 2023-09-15 下午3 57 08 | 截圖 2023-09-15 下午3 56 54 |

* Go to the password/social login page of [Gravatar](http://calypso.localhost:3000/log-in?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D943e6fc69e6b8c3d5d488f7d9e3ee0083c5700a3136cc3f25205360ef89c825f%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1), and you will the color of the links and submit button have been updated:

| Gravatar | WPJM |
| - | - |
| 截圖 2023-09-15 下午3 57 26 | 截圖 2023-09-15 下午3 56 19 |

* Login via the magic login of [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D943e6fc69e6b8c3d5d488f7d9e3ee0083c5700a3136cc3f25205360ef89c825f%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1), and wait for 90 seconds, you will see the "Resend verification email" has been shown and it should work as expected:

https://github.com/Automattic/wp-calypso/assets/21308003/6e323902-940c-4967-a29d-51e7b3885de2

* Go to the magic login page of [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D943e6fc69e6b8c3d5d488f7d9e3ee0083c5700a3136cc3f25205360ef89c825f%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1), and check these events:

**When landing on the magic login page:**
截圖 2023-09-15 下午4 14 43

**When clicking the "Sign in another way" link:**
截圖 2023-09-15 下午4 44 13

**When clicking the "Resend verification email" link:**
截圖 2023-09-15 下午4 46 19

**When clicking the "Use different email address" link:**
截圖 2023-09-15 下午4 48 18

* Go to the [password/social log page](http://calypso.localhost:3000/log-in) and [magic login](http://calypso.localhost:3000/log-in/link) page of WP.com, they should look the same.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by wellyshen** [comment]
@aaronfc I have made some improvements (based on 105749-Automattic/gravatar) and added tracking events, could you please take a look, thanks!

File: client/login/magic-login/index.jsx [hunk 1]
```diff
@@ -287,39 +335,34 @@ class MagicLogin extends Component {
...
-											}
-										) }
-									</div>
-								</div>
-							</>
-						)
-					}
+					<hr className="grav-powered-magic-login__divider grav-powered-magic-login__divider--email-form" />
+					<div className="grav-powered-magic-login__login-page-link">
+						{ translate( '{{a}}Sign in another way{{/a}}', {
```
Comment (aaronfc) [inline client/login/magic-login/index.jsx:348]: I am not completely sure but shouldn't this be "Sign-in in another way" ? "Sign in another way" reads a bit strange to me.

File: client/login/magic-login/index.jsx [hunk 2]
```diff
@@ -224,51 +264,59 @@ class MagicLogin extends Component {
...
-									},
-								}
-							) }
+							{ resendEmailCountdown === 0
+								? translate(
+										'{{sendEmailButton}}Resend the verification email{{/sendEmailButton}} or {{showMagicLoginButton}}use a different email address{{/showMagicLoginButton}}.',
+										emailTextOptions
+								  )
+								: translate(
+										'{{showMagicLoginButton}}use a different email address{{/showMagicLoginButton}}.',
```
Comment (aaronfc) [inline client/login/magic-login/index.jsx]: Maybe make the first letter capitalised? It looks a bit strange when the "Resend" link is not there.

image

**Review by aaronfc — APPROVED** [review]
Looks good and works as described!

I left a couple of comments regarding text/copies. Feel free to merge after you have considered both.

File: client/login/magic-login/index.jsx (hunk 1 above)
Comment (wellyshen) [reply client/login/magic-login/index.jsx:348]: @aaronfc  I have double-checked with AI, and the "Sign in another way" is more suitable for our case.

File: client/login/magic-login/index.jsx (hunk 2 above)
Comment (wellyshen) [reply client/login/magic-login/index.jsx]: @aaronfc Good catch, I have updated it, thanks!

**Review by aaronfc — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8967947

Thank you @wellyshen for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8967947).

**Pull request merged**